rohmu/delta/snapshot.py

```diff
@@ -258,14 +264,20 @@ def snapshot(self, *, progress: Optional[Progress] = None, reuse_old_snapshotfil
         # Create missing directories
         changes = self._snapshot_create_missing_directories(src_dirs=src_dirs, dst_dirs=dst_dirs)
         progress.add_success()
+        if progress_callback:
+            progress_callback("Creating missing directories", progress.progress_metrics())
 
         # Remove extra files
         changes += self._snapshot_remove_extra_files(src_files=src_files, dst_files=dst_files)
```

Comment:
It might be better to have the progress callback called more frequently than this. I don't know in practice what the maximum time it might be for this method to complete, but I imagine for services which experience a lot of flux, it could be high.
This could be coupled with a debouncer to avoid calling it frequently, during these loops.

Reply:
yep, added callback for _snapshot_remove_extra_files